include/boost/url/detail/impl/remove_dot_segments.ipp

```diff
@@ -183,6 +183,132 @@ remove_dot_segments(
     return dest - dest0;
 }
 
+char
+path_pop_back( string_view& s )
+{
+    if (s.size() < 3 ||
+        *std::prev(s.end(), 3) != '%')
+    {
+        char c = s.back();
+        s.remove_suffix(1);
+        return c;
+    }
+    char c = 0;
+    pct_decode_unchecked(
+        &c, &c + 1, s.substr(s.size() - 3));
+    if (c != '/')
+    {
+        s.remove_suffix(3);
+        return c;
+    }
+    c = s.back();
+    s.remove_suffix(1);
+    return c;
+};
+
+void
+pop_last_segment(
```

Comment:
This was just extracted from `normalized_path_compare` so it can be reused in the hash function. Same code.